app/javascript/packages/analytics/index.ts

```diff
@@ -17,14 +17,30 @@ interface NewRelicGlobals {
  */
 export async function trackEvent(event: string, payload?: object): Promise<void> {
   const endpoint = getConfigValue('analyticsEndpoint');
+
   if (!endpoint) {
     return;
   }
 
+  const eventJson = JSON.stringify({ event, payload });
+
+  // Favor making analytics requests using sendBeacon(), which can be prioritized
+  // appropriately by the browser and have a better chance of succeeding during page unload.
+  try {
+    const blob = new Blob([eventJson], { type: 'application/json' });
```

Comment:
This `Blob` API was new to me -- `type` is used as the `Content-type` header in the resulting request.